src/main/java/servlets/logServlet.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/logServlet")
+public class logServlet extends HttpServlet {
+
```

Comment:
You can make the code more readable by introducing an inner class for response format. Here is how to do it:

Outside of doGet() function create class:

```
static class AuthResponse {
  public String loginUrl;
  public String logoutUrl;
}
```
Then within doGet():
```
AuthResponse response = new AuthResponse();
...
if (userService.isUserLoggedIn()) {
  response.logoutUrl = userService.createLogoutURL("/");
} else {
  response.loginUrl = userService.createLoginURL("/");
}
response.getWriter().println(gson.toJson(response));
```

This way on the client side in log_button() function you can access fields directly:
```
if (logUrl.loginUrl) {
  console.log("received login URL");
} else {
  console.log("received logout URL");
}
```

Reply:
When I tried implementing this the code became very long and I was getting error decided to leave out inner class but simply code as it is instead.